refactor(lexer): simplify byte handler macros

overlookmotel opened the pull request:
Pure refactor. Simplify the byte handler macros in lexer in 3 ways:

#### 1. Remove the `const BLAH: ByteHandler = { ... };` wrappers from generated code

I don't think they're necessary, and I'm not sure why I put them there in the first place.

Before:

```rs
const UNI: ByteHandler = {
    #[expect(non_snake_case)]
    fn UNI(lexer: &mut Lexer) -> Kind {
        lexer.unicode_char_handler()
    }
    UNI
};
```

After:

```rs
#[expect(non_snake_case)]
fn UNI(lexer: &mut Lexer) -> Kind {
    lexer.unicode_char_handler()
}
```

Each byte handler still has a useful name in CodSpeed flame graphs after this change.

#### 2. Remove the `byte_handler!` macro.

`ascii_byte_handler!` and `ascii_identifier_handler!` macros remain, but they no longer use `byte_handler!` macro internally.

Removing the macro-within-macro pattern may make this code easier for AI to understand (and probably ditto for humans!).

#### 3. Shorten macro expansion

Hoist `use oxc_data_structures::assert_unchecked;` to top of file, so it doesn't need to be repeated in each macro expansion.

overlookmotel commented:
* **#13057** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/13057?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/13057?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=13057).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Boshen commented:
### Merge activity

* **Aug 13, 3:32 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
refactor(lexer): simplify byte handler macros (#13057)

Pure refactor. Simplify the byte handler macros in lexer in 3 ways:

#### 1. Remove the `const BLAH: ByteHandler = { ... };` wrappers from generated code

I don't think they're necessary, and I'm not sure why I put them there in the first place.

Before:

```rs
const UNI: ByteHandler = {
    #[expect(non_snake_case)]
    fn UNI(lexer: &mut Lexer) -> Kind {
        lexer.unicode_char_handler()
    }
    UNI
};
```

After:

```rs
#[expect(non_snake_case)]
fn UNI(lexer: &mut Lexer) -> Kind {
    lexer.unicode_char_handler()
}
```

Each byte handler still has a useful name in CodSpeed flame graphs after this change.

#### 2. Remove the `byte_handler!` macro.

`ascii_byte_handler!` and `ascii_identifier_handler!` macros remain, but they no longer use `byte_handler!` macro internally.

Removing the macro-within-macro pattern may make this code easier for AI to understand (and probably ditto for humans!).

#### 3. Shorten macro expansion

Hoist `use oxc_data_structures::assert_unchecked;` to top of file, so it doesn't need to be repeated in each macro expansion.
